pip/pip-351.md

```diff
@@ -0,0 +1,166 @@
+                    confBuilder.setSslEngineFactory(sslEngineFactory);
+                } 
+```
+
+None of these options can be currently configured when using Pulsar Test client.
+
+# Motivation
+
+<!--
+Describe the problem this proposal is trying to solve.
+
+* Explain what is the problem you're trying to solve - current situation.
+* This section is the "Why" of your proposal.
+-->
+
+As we already let users both extend authentication and use just the keystore and truststore properties to set up mTLS
+connections, without using any authentication plugin class, a lot of them might want to use this method of authentication
+during Performance Testing as well.
+
+I understand that currently mTLS (for testing purposes) can be achieved by using trust and client certificates.
+However, the issue of users extending authentication plugin classes and utilizing keystores is still not covered 
+with the current options. Therefore, I propose we make these already existing options be configured in test clients,
+increasing its usability.
+
+# Goals
+
+## In Scope
+
+Create new Arguments for the following properties, in PerformanceBaseArguments.java :
+1. useKeyStoreTls
+2. trustStoreType
+3. trustStorePath
+4. trustStorePass
+5. keyStoreType
+6. keyStorePath
+7. keyStorePass
+
+Update the code to change between TrustCerts and TrustStore based on useKeyStoreTls.
+
+<!--
+What this PIP intend to achieve once It's integrated into Pulsar.
+Why does it benefit Pulsar.
+-->
+
+[//]: # (## Out of Scope)
+
+<!--
+Describe what you have decided to keep out of scope, perhaps left for a different PIP/s.
+-->
+
+
+[//]: # (# High Level Design)
+
+<!--
+Describe the design of your solution in *high level*.
+Describe the solution end to end, from a birds-eye view.
+Don't go into implementation details in this section.
+
+I should be able to finish reading from beginning of the PIP to here (including) and understand the feature and 
+how you intend to solve it, end to end.
+
+DON'T
+* Avoid code snippets, unless it's essential to explain your intent.
+-->
+
+# Detailed Design
+
+## Design & Implementation Details
+
+<!--
+This is the section where you dive into the details. It can be:
+* Concrete class names and their roles and responsibility, including methods.
+* Code snippets of existing code.
+* Interface names and its methods.
+* ...
+-->
+
+Add the options for utilizing keystores as part of performance base arguments, along with forwarding their values
+to the client/admin builders.
+
+## Public-facing Changes
+
+<!--
+Describe the additions you plan to make for each public facing component. 
+Remove the sections you are not changing.
+Clearly mark any changes which are BREAKING backward compatability.
+-->
+
+### CLI
+
+All places we utilize Pulsar Test client, for example Pulsar-Perf will have the following new options:
+
+1. --use-keystore-tls &rarr; Default value = false, Possible values = true, false
```

Comment:
```suggestion
1. --use-keystore-tls &rarr; Default value = false
```
When you use this option, which without any value:
```
pulsar-perf produce --use-keystore-tls --keystore-path $KEYSTORE_PATH .....
```

Reply:
My bad, I mixed up with the possible values it can hold vs what users will enter. I have removed this as suggested.